### PR TITLE
React native - Adding missing Sectionlist properties

### PR DESCRIPTION
**Pull request by tdekoning**
This add the `ListFooterComponentStyle` and `ListHeaderComponentStyle` properties to the `SectionList` component. for React Native.

Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: https://reactnative.dev/docs/0.65/virtualizedlist#listfootercomponentstyle
- [x] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.

### CODE REVIEW

File: types/react-native/index.d.ts [hunk 1]
```diff
@@ -4210,11 +4210,21 @@ export interface SectionListProps<ItemT, SectionT = DefaultSectionT>
      */
     ListFooterComponent?: React.ComponentType<any> | React.ReactElement | null | undefined;
 
+    /**
+     * Styling for internal View for ListFooterComponent
+     */
+    ListFooterComponentStyle?: StyleProp<ViewStyle> | undefined;
```
Comment (ca057) [inline types/react-native/index.d.ts:4216]: As it’s a https://flow.org/en/docs/types/maybe/ it should also be null :)
```suggestion
    ListFooterComponentStyle?: StyleProp<ViewStyle> | undefined | null;
```

File: types/react-native/index.d.ts [hunk 2]
```diff
@@ -4210,11 +4210,21 @@ export interface SectionListProps<ItemT, SectionT = DefaultSectionT>
...
+
     /**
      * Rendered at the very beginning of the list.
      */
     ListHeaderComponent?: React.ComponentType<any> | React.ReactElement | null | undefined;
 
+    /**
+     * Styling for internal View for ListHeaderComponent
+     */
+    ListHeaderComponentStyle?: StyleProp<ViewStyle> | undefined;
```
Comment (ca057) [inline types/react-native/index.d.ts:4226]:
```suggestion
    ListHeaderComponentStyle?: StyleProp<ViewStyle> | undefined | null;
```

File: types/react-native/test/index.tsx [hunk 3]
```diff
@@ -667,6 +667,10 @@ export class SectionListTest extends React.Component<SectionListProps<string>, {
                     )}
                     CellRendererComponent={cellRenderer}
                     maxToRenderPerBatch={5}
+                    ListFooterComponent={null}
+                    ListFooterComponentStyle={[{ padding: 8 }, [{ backgroundColor: 'transparent' }]]}
+                    ListHeaderComponent={null}
+                    ListHeaderComponentStyle={[{ padding: 8 }, [{ backgroundColor: 'transparent' }]]}
```
Comment (ca057) [inline types/react-native/test/index.tsx:673]: Thanks for adding the tests! Shall we add some with `null`/`undefined` for the style-props as well?

**Review by ca057 — APPROVED** [review]

**Comment by tdekoning** [comment]
@ca057 I've made the adjustments. Sorry for the messy commit log, there were some issues with the CI.

**Review by ca057 — APPROVED** [review]

**Pull request merged**